apiary-metastore-listener/src/test/java/com/expedia/apiary/extensions/metastore/listener/ApiarySnsListenerTest.java

```diff
@@ -106,10 +133,32 @@ public void onCreateTable() throws MetaException {
     snsListener.onCreateTable(event);
     verify(snsClient).publish(requestCaptor.capture());
     PublishRequest publishRequest = requestCaptor.getValue();
-    assertThat(publishRequest.getMessage(),
-        is("{\"protocolVersion\":\""
-            + PROTOCOL_VERSION
-            + "\",\"eventType\":\"CREATE_TABLE\",\"dbName\":\"some_db\",\"tableName\":\"some_table\"}"));
+    assertThat(publishRequest.getMessage(), is("{\"protocolVersion\":\""
+        + PROTOCOL_VERSION
+        + "\",\"eventType\":\"CREATE_TABLE\",\"dbName\":\"some_db\",\"tableName\":\"some_table\",\"tableLocation\":\"s3://table_location\",\"tableParameters\":\"{MY_VAR_TWO=5, MY_VAR_ONE=true}\"}"));
+  }
+
+  @Test
+  public void tableParamRegexNotSet() throws MetaException {
+    environmentVariables.clear("TABLE_PARAM_FILTER");
+    ApiarySnsListener snsListener = new ApiarySnsListener(configuration, snsClient);
+
+    CreateTableEvent event = mock(CreateTableEvent.class);
+    when(event.getStatus()).thenReturn(true);
+    when(event.getTable()).thenReturn(table);
+
+    snsListener.onCreateTable(event);
+    verify(snsClient).publish(requestCaptor.capture());
+    PublishRequest publishRequest = requestCaptor.getValue();
+    assertThat(publishRequest.getMessage(), is("{\"protocolVersion\":\""
+        + PROTOCOL_VERSION
+        + "\",\"eventType\":\"CREATE_TABLE\",\"dbName\":\"some_db\",\"tableName\":\"some_table\",\"tableLocation\":\"s3://table_location\",\"tableParameters\":\"{}\"}"));
+  }
+
+  @Test(expected = PatternSyntaxException.class)
+  public void tableParamRegexIsInvalid() throws MetaException {
+    environmentVariables.set("TABLE_PARAM_FILTER", "[");
+    new ApiarySnsListener(configuration, snsClient);
   }
```

Comment:
We need just another test here of the `isEmpty` logic - set the filter to "" or "    ".

Reply:
I deliberately didn't add a test for that case coz its a valid regex. It compiles just fine but won't match any of the table params. So, an empty tableParams is returned.

Do you think we should throw an exception for this also ?

Reply:
Hmmm, like you say it's a valid regex so another option would be not too do the `isEmpty` check but instead only check for null on the string and allow people to set blank strings. The other option would be to throw an exception on a blank string but then we need to document that you can't set a regex that is a blank string. Another one for the meeting tomorrow to get a decision on!

Reply:
I have added a unit test for this scenario as well. I could have also used `StringUtils.isBlank()` to just ignore blank regex but I preferred to treat it just like any other regex as its a valid regex. I have added a log statement in the constructor to print the TABLE_PARAM_FILTER regex for faster discovery of any such issue.

Reply:
:+1: